cci_nepal/analysis/wp2_data_scoping/database_sample/database_sample_descriptive.py

```diff
@@ -0,0 +1,190 @@
+
+# %%
+db.drop(db.columns[2], axis=1, inplace=True)
+
+
+# %%
+def clean_df_columns(df):
+    df.columns = df.columns.str.replace("[^a-zA-Z0-9\-\s]+", "", regex=True)
+    df.columns = df.columns.str.lstrip()
+
+
+# %%
+clean_df_columns(db)
+
+# %%
+db = db.replace("[^a-zA-Z0-9\-\s]+", "", regex=True)
+
+# %%
+db.head(5)
+
+# %%
+percent_missing = db.isnull().sum() * 100 / len(db)
+missing_value_df = pd.DataFrame(
+    {"column_name": db.columns, "percent_missing": percent_missing}
+)
+missing_value_df.sort_values("percent_missing", inplace=True)
+
+# %%
+missing_value_df["percent_missing"].plot(kind="bar")
+plt.xticks([])
+
+# %%
+plt.figure(figsize=(15, 10))
+missing_value_df["percent_missing"].tail(60).plot(kind="bar")
+
+# %%
+# db.to_excel('clean-temp.xlsx', index=False)
+
+# %%
+db["Ethnicity of the informant"] = (
+    db["Ethnicity of the informant"].str.replace("\d+", "").copy()
+)
+
+# %%
+fig = plt.figure()
+db["Ethnicity of the informant"].value_counts().sort_values().plot(kind="barh")
+fig.suptitle("Ethnicity of the informant", fontsize=14)
+plt.xlabel("Frequency", fontsize=12)
+plt.ylabel("Ethnicity", fontsize=12)
+
+# %%
+nfri_items = {}
+nfri_df = db.iloc[:, 52:69]
+nfri_df.columns = nfri_df.columns.str.replace(
+    "What did your family get from Nepal Red Cross Society after the flood and landslide",
+    "",
+)
+nfri_df.columns = nfri_df.columns.str.strip()
+
+# %%
+for col in nfri_df.columns:
+    nfri_df[col] = nfri_df[col].fillna(0)
+    nfri_items[col] = sum(nfri_df[col])
+
+# %%
+fig = plt.figure()
+
+plt.bar(nfri_items.keys(), nfri_items.values())
+plt.xticks(rotation=90)
+plt.tight_layout()
+
+fig.suptitle(
+    "What did your family get from Nepal Red Cross Society after the flood and landslide?",
+    fontsize=14,
+)
+plt.xlabel("Items", fontsize=12)
+plt.ylabel("Frequency", fontsize=12)
+
+plt.show()
+
+# %%
+fig = plt.figure(figsize=(10, 15))
+db.iloc[:, 69].value_counts().sort_values().plot(kind="barh")
+fig.suptitle("Other relief items specified", fontsize=14)
+plt.xlabel("Frequency", fontsize=12)
+plt.ylabel("Items", fontsize=12)
+
+# %%
+cash_relief = {}
+cr_df = db.iloc[:, 90:99]
+cr_df.columns = cr_df.columns.str.replace(
+    "For what did you spend the money you received from the Red Cross", ""
+)
+cr_df.columns = cr_df.columns.str.strip()
+
+# %%
+for col in cr_df.columns:
+    cr_df[col] = cr_df[col].fillna(0)
+    cash_relief[col] = sum(cr_df[col])
+
+# %%
+fig = plt.figure()
+
+plt.bar(cash_relief.keys(), cash_relief.values())
+plt.xticks(rotation=90)
+
+fig.suptitle("How cash relief was spent", fontsize=14)
+plt.xlabel("Spending activity", fontsize=12)
+plt.ylabel("Frequency", fontsize=12)
+
+plt.show()
+
+# %%
+fig = plt.figure()
+db[
+    "Did the cash relief provided by the Red Cross met your immediate needs"
+].value_counts().sort_values().plot(kind="barh")
+plt.xticks(rotation=90)
+fig.suptitle(
+    "Did the cash relief provided by the Red Cross met your immediate needs?",
+    fontsize=14,
+)
```

Comment:
It maybe good to save useful figures in outputs directory so that they can be viewed given that the notebooks are not submitted? - following recommendation from the cookiecutter doc.

Reply:
True! Will also update so the figures are saved :)